test/test.go

```diff
@@ -0,0 +1,265 @@
+package test
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/bitrise/models"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-steplib/steps-deploy-to-bitrise-io/test/converters"
+)
+
+// FileInfo ...
+type FileInfo struct {
+	FileName string `json:"filename"`
+	FileSize int    `json:"filesize"`
+}
+
+// UploadURL ...
+type UploadURL struct {
+	FileName string `json:"filename"`
+	URL      string `json:"upload_url"`
+}
+
+// UploadRequest ...
+type UploadRequest struct {
+	Name   string                    `json:"name"`
+	Step   models.TestResultStepInfo `json:"step_info"`
+	Assets []FileInfo                `json:"assets"`
+	FileInfo
+}
+
+// UploadResponse ...
+type UploadResponse struct {
+	ID     string      `json:"id"`
+	Assets []UploadURL `json:"assets"`
+	UploadURL
+}
+
+// Result ...
+type Result struct {
+	Name       string
+	XMLContent []byte
+	ImagePaths []string
+	StepInfo   models.TestResultStepInfo
+}
+
+// Results ...
+type Results []Result
+
+func httpCall(apiToken, method, url string, input io.Reader, output interface{}) error {
+	if apiToken != "" {
+		url = url + "/" + apiToken
+	}
+	req, err := http.NewRequest(method, url, input)
+	if err != nil {
+		return err
+	}
+
+	b, err := httputil.DumpRequest(req, true)
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(b))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	b, err = httputil.DumpResponse(resp, true)
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(b))
+
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			fmt.Printf("Failed to close body, error: %s\n", err)
+		}
+	}()
+
+	if resp.StatusCode < 200 || 299 < resp.StatusCode {
+		bodyData, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("unsuccessful response code: %d and failed to read body, error: %s", resp.StatusCode, err)
+		}
+		return fmt.Errorf("unsuccessful response code: %d\nbody:\n%s", resp.StatusCode, bodyData)
+	}
+
+	if output != nil {
+		return json.NewDecoder(resp.Body).Decode(&output)
+	}
+	return nil
+}
+
+func findImages(testDir string) (imageFilePaths []string) {
+	for _, ext := range []string{".jpg", ".jpeg", ".png"} {
+		if paths, err := filepath.Glob(filepath.Join(testDir, "*"+ext)); err == nil {
```

Comment:
Do not forget, that the glob uses a pattern:
https://golang.org/pkg/path/filepath/#Match

Fix:
https://github.com/bitrise-io/go-xcode/blob/b3f7172f910693d60973a07b61db757aca81f778/utility/glob.go#L4